### ISSUE TITLE
Investigate mjsunit/wasm/embenchen/zlib.js r6 simulator timeout failures

### ISSUE THREAD

**Issue by kisg** [v8mips/v8mips#290]

**Comment by kilvadyb** [comment]
mjsunit/wasm/embenchen/zlib sometimes TIMEOUTs on r6 simulator. I run the test on the same (otherwise idle) PC and on r2 sim I got:

```
real    0m56.528s
user    0m55.482s
sys    0m0.060s
```

and on r6 sim:

```
real    0m59.056s
user    0m58.600s
sys    0m0.036s
```

So it looks like the test is slightly slower on r6 sim then on r2 sim and is around the TIMEOUT limit. It passes otherwise (without time checking it is always fine).

I did some profiling running this test and cannot see any serious slowdown in r6 simulator compared to r2 simulator. I asked Brandon to increase the r6 sim timeout on the buildbots and since then the failure seems to disappear from the bots.

The mjsunit/wasm/embenchen/zlib test is a huge and slow one so I am going to do some full test time measurement to find out if there are other speed differences and are there any simpler test case to investigate.
